cli/auth.go

```diff
@@ -112,20 +123,23 @@ func (c *viamClient) loginAction(cCtx *cli.Context) error {
 			already = ""
 			viamLogo(cCtx.App.Writer)
 		}
-
 		fmt.Fprintf(cCtx.App.Writer, "%slogged in as %q, expires %s\n", already, t.User.Email,
 			t.ExpiresAt.Format("Mon Jan 2 15:04:05 MST 2006"))
 	}
 
-	if c.conf.Auth != nil && !c.conf.Auth.isExpired() {
-		loggedInMessage(c.conf.Auth, true)
+	currentToken, ok := c.conf.Auth.(*token)
+	if !ok {
+		return errors.New("config auth was not a token or api key")
+	}
```

Comment:
[bug] What do we think the expected behavior is here? Should it delete their api key?

Reply:
It looks like you are only casting to a `*token`, don't we also want to check if there's an API key? In any case, I think you should `logout` so that the cache is cleared.